enhancements/console/internationalization.md

```diff
@@ -0,0 +1,158 @@
+
+We will need to mark all hard-coded strings with react-i18next so they can be extracted using [i18next-parser](https://github.com/i18next/i18next-parser) and handed off to the Globalization team.
+This is not complex work but will involve many small code changes.
+[A rough string search](https://gist.github.com/jschuler/a404aed1ea774383cc2decd57d3135c6)
+of the frontend console code found about 9,000 strings, but many will not be translated (such as Kubernetes resource names).
+We will need to identify a prioritized list of UIs in OpenShift for localization.
+
+We will also need to do testing, coordinate with the Red Hat Globalization team on testing and translation,
+and collaborate with console plugins on a localization strategy for releases. Finally, a product testing window will need to be defined.
+
+### Implementation Details/Notes/Constraints [optional]
+
+Localization will affect multiple teams within the console development space.
+This will affect Dev console, Admin console, and other packages like CNV.
+The solution must account for the plugin contribution model and enable dynamic extensions to provide localization as necessary.
+
+We plan to break translations up into [multiple translation files](https://react.i18next.com/guides/multiple-translation-files), at least one per package.
+This is expected to include all information we can localize in the console, potentially broken over multiple phases.
+As mentioned earlier in this document, localization of all text within the system is not possible. In addition, some terms are Kubernetes terms and would not be translated.
+
+Some components may need to be updated to accommodate longer text lengths.
+
+The font system seems to handle Japanese and Chinese double-byte characters well.
+
+### Risks and Mitigations
+
+**Risk**: All text is not translated.
+
+**Mitigation**: Work with the UXD team on a design strategy.
+
+**Risk**: Translations are incorrect or don't work as expected.
+
+**Mitigation**: We will work with the Red Hat Globalization team to thoroughly test the application and obtain culturally sensitive and accurate translations.
+We will also write integration tests to ensure that the i18n-next translation framework is working as expected.
+
+**Risk**: Components may not be able to accommodate longer text lengths seen in other languages.
+
+**Mitigation**: We have added a plugin that generates pseudotranslations, which will allow us to test components against longer text lengths and make fixes as needed.
+
+## Design Details
+
+### Open Questions [optional]
+
+* Will translations be done centrally for all plugins or will every team needs to have a separate team of translators?
+Setting up a team of translators specifically for kubevirt might be quite a challenge.
+* What is the desired fallback experience if not all pieces/parts of the UI are translated for a given language?
```

Comment:
Typically the translation library will simply fallback to the default language string; in our case english.